addOns/postman/src/main/java/org/zaproxy/addon/postman/PostmanParser.java

```diff
@@ -176,20 +205,42 @@ public PostmanCollection parse(String collectionJson) throws JsonProcessingExcep
     }
 
     static void extractHttpMessages(List<AbstractItem> items, List<HttpMessage> httpMessages) {
-        extractHttpMessages(items, httpMessages, new ArrayList<>());
+        extractHttpMessages(items, httpMessages, new ArrayList<>(), null);
     }
 
     static void extractHttpMessages(
-            List<AbstractItem> items, List<HttpMessage> httpMessages, List<String> errors) {
+            List<AbstractItem> items,
+            List<HttpMessage> httpMessages,
+            List<String> errors,
+            List<KeyValueData> parentVariables) {
         if (items != null) {
             for (AbstractItem item : items) {
                 if (item instanceof Item) {
-                    HttpMessage httpMessage = extractHttpMessage((Item) item, errors);
+                    HttpMessage httpMessage =
+                            extractHttpMessage((Item) item, errors, parentVariables);
                     if (httpMessage != null) {
                         httpMessages.add(httpMessage);
                     }
                 } else if (item instanceof ItemGroup) {
-                    extractHttpMessages(((ItemGroup) item).getItem(), httpMessages, errors);
+                    ItemGroup itemGroup = (ItemGroup) item;
+
+                    // parent's and own variables, to be given to the children
+                    List<KeyValueData> allVariables = new ArrayList<>();
+
+                    if (parentVariables != null) {
+                        allVariables.addAll(parentVariables);
+                    }
+
+                    List<KeyValueData> itemGroupVariables = itemGroup.getVariable();
+                    if (itemGroupVariables != null) {
+                        allVariables.addAll(itemGroupVariables);
+                    }
```

Comment:
I think its needed to create the new list, as `itemGroupVariables` is unmodifiable and we should not add to `parentVariables` as it can disturb the logic.
Will push with the extracted method.

Reply:
Right, but if `itemGroupVariables` is `null`/`empty` we could return `parentVariables` or vice versa. Though not a necessity for this PR.